app.js

```diff
@@ -165,46 +165,152 @@ $(document).ready(function () {
-            function initializeSearch() {
-                currentLocation = new google.maps.LatLng(latitude, longitude);
-                map = new google.maps.Map(document.getElementById("map"), {
-                    center: currentLocation,
-                    zoom: 13
-                });
-                var searchInput = $("#SearchField")
-                    .val()
-                    .trim()
-                    .toLowerCase();
-
-                request = {
-                    location: currentLocation,
-                    radius: "400",
-                    query: searchInput + " cocktail"
-                };
-                service = new google.maps.places.PlacesService(map);
-                service.textSearch(request, callback);
+        if ($("#SearchField").val().length === 0) {
+            console.log("Modal should open");
+            // Get the modal
+            var modal = document.getElementById("myModal");
+
+            // Get the <span> element that closes the modal
+            var span = document.getElementsByClassName("close")[0];
+
+            // When the user clicks the button, open the modal 
+
+            modal.style.display = "block";
+
+
+            // When the user clicks on <span> (x), close the modal
+            span.onclick = function () {
+                modal.style.display = "none";
+            }
+
+            // When the user clicks anywhere outside of the modal, close it
+            window.onclick = function (event) {
+                if (event.target == modal) {
+                    modal.style.display = "none";
+                }
+            }
+        }
+        else {
+            if ($("#barOption").is(":checked")) {
+                // Function to take the user search input and display results on the map
+                function initializeSearch() {
+                    currentLocation = new google.maps.LatLng(latitude, longitude);
+                    map = new google.maps.Map(document.getElementById("map"), {
+                        center: currentLocation,
+                        zoom: 13
+                    });
+                    var searchInput = $("#SearchField")
+                        .val()
+                        .trim()
+                        .toLowerCase();
+
+                    request = {
+                        location: currentLocation,
+                        radius: "400",
+                        query: searchInput + " cocktail"
+                    };
+                    service = new google.maps.places.PlacesService(map);
+                    service.textSearch(request, callback);
+                }
+                initializeSearch();
+                $("#SearchField").val("");
+            } else if ($("#drinkOption").is(":checked")) {
+                var cocktailSearch = $("#SearchField").val();
+                drinkSearch(cocktailSearch);
+            }
+            else {
```

Comment:
Line 217 is handling the drink option search and calling the drinkSearch function on 219. delete lines 221-300

Reply:
The else on 193 encompasses all of your original code- (the else on 193 triggers when there is input in the search bar).  The if on 168 triggers the modal (when there is no input in the search bar)

Reply:
After thinking about it more, I think you are a correct because there are only 2 options, input is blank or its not blank. If it's not blank then it runs my code and checks which radio button is selected. I was originally thinking there were 3 options which were 1) input is blank, 2) input is not blank and barOption is checked, and 3 input is not blank and drinkOption is checked. Lines 221-300 are redundant though and don't need to be added.

Reply:
Gotcha, ok- I'm taking out 221-300 and recommitting, thanks